make-prs.py

```diff
@@ -0,0 +1,251 @@
+import argparse
+import logging
+import os
+import re
+import shutil
+import subprocess
+import typing
+
+import github
+import requests
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+WORKING_DIR = "./tmp"
+DIFF_FILE = "./diff.patch"
+
+auth = github.Auth.Token(os.getenv("GITHUB_TOKEN"))
+gh_client = github.Github(auth=auth)
+
+
+def get_diff_as_patch(repo: str, pull_request_number: int):
+    logger.info("Getting diff from PR ...")
+
+    pr_diff_url = gh_client.get_repo(repo).get_pull(pull_request_number).diff_url
+
+    res = requests.get(pr_diff_url)
+    if res.status_code != requests.codes.ok:
+        logger.error(f"Error getting diff from PR {pull_request_number}.")
+        exit(1)
+
+    with open(DIFF_FILE, "w") as f:
+        f.write(res.text)
+
+
+def clone(repo: str):
+    logger.info("Cloning repository ...")
+
+    shutil.rmtree(WORKING_DIR, ignore_errors=True)
+
+    clone_url = f"git@github.com:{repo}.git"
+    command = ["git", "clone", clone_url, WORKING_DIR]
+
+    result = subprocess.run(command, capture_output=True, text=True)
+    if result.returncode != 0:
+        logger.error(f"Error cloning reop {repo}: {result.stderr}")
+        exit(1)
+
+
+def create_new_branch_based_on(base: str) -> str:
+    logger.info("Creating branch ...")
+
+    head = f"{base}-update"
+
+    command = [
+        "git",
+        "checkout",
+        "-b",
+        head,
+        f"origin/{base}",
+    ]
+    result = subprocess.run(command, cwd=WORKING_DIR, capture_output=True, text=True)
+    if result.returncode != 0:
+        logger.error(f"Error creating new branch {head}: {result.stderr}")
+        exit(1)
+
+    return head
+
+
+def apply_patch() -> bool:
+    logger.info("Applying patch ...")
+
+    command = [
+        "git",
+        "apply",
+        "--3way",
+        "../diff.patch",
+    ]
+    result = subprocess.run(command, cwd=WORKING_DIR, capture_output=True, text=True)
+    if result.returncode != 0:
+        if "conflicts" in result.stderr:
+            return True
+        else:
+            logger.error(f"Error applying diff: {result.stderr}")
+            exit(1)
+
+
+def commit_diff(head: str):
+    logger.info("Pushing changes ...")
+
+    add = ["git", "add", "."]
+    subprocess.run(add, cwd=WORKING_DIR, capture_output=True, text=True)
+
+    commit = ["git", "commit", "-m", "chore: merging diff"]
+    subprocess.run(commit, cwd=WORKING_DIR, capture_output=True, text=True)
+
+    push = ["git", "push", "-f", "origin", head]
+    result = subprocess.run(push, cwd=WORKING_DIR, capture_output=True, text=True)
+
+    if result.returncode != 0:
+        logger.error(f"Error pushing changes to branch {head}: {result.stderr}")
+        exit(1)
+
+
+def create_pr(repo: str, base: str, head: str, conflict: bool):
+    logger.info("Creating PR ...")
+
+    title = f"chore: merging diff into branch {base}"
+    body = f"Automated change: merging diff into branch {base}"
```

Comment:
Could we make these options to the script? These would be fine as default values, but I think it would be handy to actually pass them into the script rather than edit the PR later.

The default probably should have the original reference PR linked somewhere too, so you know what changes are coming through (maybe that's actually enough rather than needing them to be options).